books_crud/views.py

```diff
@@ -0,0 +1,163 @@
+from django.shortcuts import get_object_or_404
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+from .models import BookShelf, Author
+from .serializers import BookSerializer
+import requests
+from .utils import get_the_index_of_author
+
+
+@api_view(['GET'])
```

Comment:
You can also use generic ViewSets, that will handle generics actions automatically. Create a new commit with using ViewSet to practice them

Reply:
I made these changes and pushed another commit.

Reply:
@ejazahmad122 create a new commit with using Class based Generic Views as well.
https://docs.djangoproject.com/en/4.1/topics/class-based-views/generic-display/